src/main/java/edu/wisc/my/personalizedredirection/controller/PersonalizedRedirectionController.java

```diff
@@ -79,7 +82,13 @@ public void setRedirectionService(IRedirectionService redirectionService) {
             logger.error("Issues happened while trying to generate custom link",
                     e);
             response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            hasError = true;
          }
+
+         if(hasError){
+            response.sendRedirect(errorURL);
```

Comment:
It's weird that it's not picking up our default error page.  I'm going to give something a quick try.